spec/features/ingredients/new_ingredient_page_spec.rb

```diff
@@ -0,0 +1,19 @@
+feature 'user visits new ingredient page' do
+  before do
+    visit('/ingredients/new')
+  end
+
+  it 'creates new ingredient' do
+    fill_in('ingredient_name', with: 'Example Ingredient')
+    fill_in('ingredient_unit_type', with: 'Type')
+    click_button('Save')
+    current_path.match(ingredients_path)
+  end
+
+  it 'fails to create new ingredient' do
```

Comment:
`shows validation errors` - you should describe the desired result, not the context. Ideally it should be constructed like that:

```ruby
context 'when invalid params are submitted' do
  before do
    fill_in('ingredient_name', with: '')
    fill_in('ingredient_unit_type', with: '')
    click_button('Save')
  end

  it 'displays validation errors' do
    current_path.match(new_ingredient_path)
    expect(page).to have_content('is not unique')
  end
end
```

So we have:

1. `context` - environmental changes that have final impact on the output.
2. `it` - assert for desired output